backend/routes.py

```
@@ -37,59 +42,83 @@ def register_event_handlers():
+        # Verify GitHub webhook signature
+        secret = os.getenv("GITHUB_WEBHOOK_SECRET")
+        if secret:
+            expected = "sha256=" + hmac.new(
+                secret.encode(),
+                raw_body,
+                hashlib.sha256
+            ).hexdigest()
+            signature = request.headers.get("X-Hub-Signature-256")
+            if not signature or not hmac.compare_digest(expected, signature):
+                logger.warning("Invalid GitHub webhook signature")
+                raise HTTPException(status_code=401, detail="Invalid webhook signature")
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)